### PR TITLE
Update everything to match MolSSI cookiecutter

### PR DESCRIPTION
**Pull request by jchodera**
This PR overhauls the testing infrastructure to match the current [MolSSI cookiecutter](https://github.com/MolSSI/cookiecutter-cms).

`logsumexp` has been moved form `scipy.misc` to `scipy.special`.

cc: https://github.com/choderalab/yank/pull/1162

 - [x] Implement feature / fix bug
 - [x] Add [tests](https://github.com/choderalab/openmmtools/tree/master/openmmtools/tests)
 - [x] Update [documentation](https://github.com/choderalab/openmmtools/tree/master/docs) as needed
 - [x] Update [changelog](https://github.com/choderalab/openmmtools/blob/master/docs/releasehistory.rst)

### CODE REVIEW

**Comment by jchodera** [comment]
The one deviation (I know of) is that I'm still using `nosetests` here since it would take considerable effort to convert all the test generators over to `pytest` right now.

**Comment by jaimergp** [comment]
I haven't tried, but `pytest` [should be able to run](https://docs.pytest.org/en/latest/nose.html) `nosetests` just fine.

**Comment by andrrizzi** [comment]
The main problem for the conversions will be the many nose generators using `yield`. As far as I understand, `pytest`'s support for those is shaky.

File: devtools/conda-envs/test_env.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  - libnetcdf >=4.6.2 # workaround for libssl issues
+  - pyyaml
+  - cython
+  - sphinxcontrib-bibtex
+  - mpiplus
+  - pymbar
+  - pyyaml
+
+    # Testing
+  - nose
```
Comment (andrrizzi) [inline devtools/conda-envs/test_env.yaml]: I think we'll have to add also `nose-timer` here to get the `--with-timer` option working.

**Comment by jchodera** [comment]
@andrrizzi : Interestingly, we're now seeing segfaults in a different place!
```
Test method AlchemicalState.apply_to_context. ... python(2864,0x7fff8a436340) malloc: *** error for object 0x7fb943705c38: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```

**Pull request merged**